sous-chef/tests/integration_tests/menu/create_menu/test_create_menu_integration.py

```diff
@@ -1,136 +1,96 @@
-
+class TestMenuGsheets:
     @staticmethod
-    def test_finalize_fixed_menu_fails_for_record_exception(
-        menu, menu_config, mock_recipe_book
+    def test_fill_menu_template_and_load_final_menu(
+        menu_with_recipe_book,
+        menu_config,
+        frozen_due_datetime_formatter,
+        gsheets_helper,
+        fixed_final_menu,
     ):
-        menu_config.fixed.menu_number = 0
-        menu.tuple_log_exception = (RecipeNotFoundError,)
+        with patch.object(
+            DueDatetimeFormatter,
+            "anchor_datetime",
+            new_callable=PropertyMock,
+            return_value=frozen_due_datetime_formatter.anchor_datetime,
+        ):
+            final_menu_df = menu_with_recipe_book.fill_menu_template()
 
-        mock_recipe_book.get_recipe_by_title.side_effect = RecipeNotFoundError(
-            recipe_title="dummy", search_results="dummy"
-        )
-
-        # derived exception MenuIncompleteError
-        with pytest.raises(Exception) as error:
-            menu.finalize_fixed_menu()
-
-        assert (
-            str(error.value)
-            == "[menu had errors] will not send to finalize until fixed"
-        )
-        assert set(menu.record_exception) == {
-            "[recipe not found] recipe=dummy search_results=[dummy]"
-        }
-
-    @staticmethod
-    def test_get_menu_for_grocery_list(menu_with_recipe_book):
-        menu_with_recipe_book.get_menu_for_grocery_list()
-
-    @staticmethod
-    def test_get_menu_for_grocery_list_fails_for_record_exception(
-        menu, menu_config, mock_recipe_book, capsys
-    ):
-        menu_config.errors["recipe_not_found"] = "log"
-        mock_recipe_book.get_recipe_by_title.side_effect = RecipeNotFoundError(
-            recipe_title="dummy", search_results="dummy"
-        )
-
-        # derived exception MenuIncompleteError
-        with pytest.raises(Exception) as error:
-            menu.get_menu_for_grocery_list()
-
-        assert (
-            str(error.value)
-            == "[menu had errors] will not send to grocery list until fixed"
-        )
-        assert (
-            "[recipe not found] recipe=dummy search_results=[dummy]"
-            in capsys.readouterr().out
-        )
+        assert_equal_dataframe(final_menu_df, fixed_final_menu)
 
     @staticmethod
-    @pytest.mark.todoist
-    def test__upload_menu_to_todoist(menu, todoist_helper):
-        final_menu_df = menu._load_final_menu()
-        menu._upload_menu_to_todoist(
-            final_menu_df=final_menu_df, todoist_helper=todoist_helper
-        )
-        with patch("builtins.input", side_effect=[YesNoChoices.yes.value]):
-            todoist_helper.delete_all_items_in_project(project=PROJECT)
+    def test_load_final_menu(menu, gsheets_helper, fixed_final_menu):
+        final_menu_df = menu.load_final_menu(gsheets_helper=gsheets_helper)
+        assert_equal_dataframe(final_menu_df, fixed_final_menu)
+
+    # @staticmethod
```

Comment:
@ArBridgeman todo maybe more in another PR

Reply:
remove these lines as functionality already tested in add_current_menu_to_history

Reply:
@AkagiShigeru will do